yts/test_suite_test.go

```diff
@@ -123,110 +123,135 @@ func runTest(t *testing.T, testPath string) {
+				"Test: %s\nDescription: %s\nError: Marshal output doesn't match canonical\n"+
+					"Expected: %+v\nGot     : %+v",
+				testPath, testDescription, string(expectedOutYAML), string(marshaledYAML))
 		}
-		expectedOutYAML, err := os.ReadFile(outYAMLPath)
-		if err != nil {
-			t.Errorf("Test: %s\nDescription: %s\nError: Failed to read out.yaml: %v", testPath, testDescription, err)
+	})
+
+	t.Run("RoundtripTest", func(t *testing.T) {
+		shouldSkipTest(t)
+		var unmarshaledValue interface{}
+
+		if expectError {
 			return
 		}
-		var expectedUnmarshaledValue any
-		err = yaml.Unmarshal(expectedOutYAML, &expectedUnmarshaledValue)
-		if err != nil {
-			t.Errorf("Test: %s\nDescription: %s\nError: Failed to unmarshal out.yaml: %v", testPath, testDescription, err)
-			return
+
+		if err := yaml.Unmarshal(inYAML, &unmarshaledValue); err != nil {
+			t.Errorf(
+				"Test: %s\nDescription: %s\nError: Failed to unmarshal in.yaml: %v",
+				testPath, testDescription, err)
+		}
+
+		marshaledYAML, marshalErr := yaml.Marshal(unmarshaledValue)
+		if marshalErr != nil {
+			t.Errorf(
+				"Test: %s\nDescription: %s\nError: Failed to marshal value: %v",
+				testPath, testDescription, marshalErr)
 		}
-		var reUnmarshaledValue any
-		err = yaml.Unmarshal(marshaledYAML, &reUnmarshaledValue)
-		if err != nil {
-			t.Errorf("Test: %s\nDescription: %s\nError: Failed to re-unmarshal marshaled YAML: %v", testPath, testDescription, err)
-		} else if !reflect.DeepEqual(reUnmarshaledValue, expectedUnmarshaledValue) {
-			t.Errorf("Test: %s\nDescription: %s\nError: Marshal output mismatch\nExpected: %+v\nGot     : %+v", testPath, testDescription, expectedUnmarshaledValue, reUnmarshaledValue)
+
+		var roundtrippedValue interface{}
+		if err := yaml.Unmarshal(marshaledYAML, &roundtrippedValue); err != nil {
+			t.Errorf(
+				"Test: %s\nDescription: %s\nError: Failed to roundtrip in.yaml: %v",
+				testPath, testDescription, err)
+		} else if !reflect.DeepEqual(unmarshaledValue, roundtrippedValue) {
+			t.Errorf(
+				"Test: %s\nDescription: %s\nError: YAML unmarshal vs roundtripped unmarshal mismatch\n"+
+					"Expected: %+v\nGot     : %+v",
+				testPath, testDescription, unmarshaledValue, roundtrippedValue)
 		}
 	})
 
 	t.Run("JSONComparisonTest", func(t *testing.T) {
 		shouldSkipTest(t)
-		var currentUnmarshaledValue any
 
-		currentUnmarshalErr := yaml.Unmarshal(inYAML, &currentUnmarshaledValue)
-
-		if !(currentUnmarshalErr == nil || expectError) {
-			return
-		}
-		inJSONPath := filepath.Join(testPath, "in.json")
-		if _, err := os.Stat(inJSONPath); err != nil {
+		// Nothing to test if unmarshall is expected to error
+		if expectError || !fileExists(testPath, "in.json") {
 			return
 		}
-		inJSON, err := os.ReadFile(inJSONPath)
-		if err != nil {
-			t.Errorf("Test: %s\nDescription: %s\nError: Failed to read in.json: %v", testPath, testDescription, err)
-			return
+		inJSON := mustRead(t, testPath, "in.json")
+
+		var unmarshaledValue interface{}
```

Comment:
```suggestion
		var unmarshaledValue any
```
`any` is more readable and way shorter :)